extension/src/functions.rs

```diff
@@ -222,27 +248,54 @@ impl<Runtime: pallet_contracts::Config> Function for Tuple {
 #[cfg(test)]
 mod tests {
 	use super::*;
-	use crate::mock::{DispatchCallFuncId, Environment, Ext, ReadStateFuncId};
+	use crate::mock::{Environment, Ext, ReadStateEverthingFuncId};
 	use codec::Encode;
-	use mock::RuntimeRead;
+	use frame_support::traits::{Everything, Nothing};
+	use matching::WithFuncId;
+	use mock::{RuntimeRead, Test};
+	use sp_core::ConstU32;
+
+	enum AtLeastOneByte {}
+	impl Contains<Vec<u8>> for AtLeastOneByte {
+		fn contains(input: &Vec<u8>) -> bool {
+			input.len() > 0
+		}
+	}
+
+	type TestFunctions = (
```

Comment:
@Daanvdplas Even though you already told me these might not be necessary, I think it is cleaner to have it for testing purpose because we only want to check the Filtering and the execution of function with filtered functions. Let me know your thoughts on this. Already simplify the code.